src/julia/tests/test_tools.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+import platform
+import sysconfig
+import os
+import glob
+
+from julia.tools import julia_py_executable
+
+
+def julia_py_with_command_extension():
+    extension = ".cmd" if platform.system() == "Windows" else ""
+    return "julia-py" + extension
+
+
+def glob_mock(path):
+    # we're only handling the case when the glob is ".../julia-py*" or just "julia-py*"
+    # if dirname(path) is empty then return empty list - this is indicator that we shouldn't "find" any files matching the glob
+    return [] if os.path.dirname(path) == "" else [os.path.join(os.path.dirname(path), julia_py_with_command_extension())]
+
+
+def test_find_julia_py_executable_by_scheme(monkeypatch):
+    # could extend this to test different kinds of scheme
+    # right now we just fake the "posix_user" scheme and don't use the rest of get_path_mock_scheme()
+    fake_user = "fa9a5150-8e17-11ea-3f8d-ff1e5ae4a251"
+    fake_folder = "ce410603-12fa-42a2-9a24-e44cd94f6968"
+
+    def get_path_mock_scheme(scheme_match, scheme):
+        # sample scripts paths for each scheme type as reported by sysconfig.get_path("scripts",scheme) on a Linux system
+        # sysconfig.get_path("scripts","nt_user") is going to have a different form on Windows, something like "\user\xyz\appdata\..."
+        if scheme == scheme_match:
+            if scheme == "posix_user" or scheme == "osx_framework_user":
+                return os.path.join(os.path.abspath("home"), fake_user, ".local", "bin")
```

Comment:
I'm not sure why you are using `os.path.abspath`.  Maybe you mean `os.path.join(os.sep, "home", fake_user, ...)`?

Reply:
I called `os.path.abspath("home")` etc. to do just that, get `/home` to start the path off without depending on the particular symbol on *nix and MacOS, but wasn't sure whether Windows would want to stick in a drive letter at the start also. Anyway will just use `os.sep` .